homeassistant/components/twinkly/light.py

```diff
@@ -185,16 +185,47 @@ async def async_turn_on(self, **kwargs: Any) -> None:
                 if isinstance(self._attr_rgbw_color, tuple):
 
                     await self._client.interview()
-                    # Static color only supports rgb
-                    await self._client.set_static_colour(
-                        (
-                            self._attr_rgbw_color[0],
-                            self._attr_rgbw_color[1],
-                            self._attr_rgbw_color[2],
+                    if LightEntityFeature.EFFECT in self.supported_features:
+                        # Static color only supports rgb
+                        try:
+                            await self._client.set_static_colour(
+                                (
+                                    self._attr_rgbw_color[0],
+                                    self._attr_rgbw_color[1],
+                                    self._attr_rgbw_color[2],
+                                )
+                            )
+                            await self._client.set_mode("color")
+                            self._client.default_mode = "color"
+                        except ClientResponseError as error:
+                            if error.status == HTTPNotFound().status:
+                                self._attr_supported_features = (
+                                    self.supported_features & ~LightEntityFeature.EFFECT
+                                )
+                                await self._client.set_cycle_colours(
```

Comment:
This is an exact copy of the code of when the feature is not supported. That's not good. Better to just call yourself again.

```python
await self.async_turn_on(call)
return
```

Reply:
Removed as we now rely on software-version instead of http-errors to detect supported features